common-lib/src/main/java/com/google/cloud/tools/intellij/util/ProjectUtil.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.util;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.util.messages.Topic;
+import java.util.stream.Stream;
+import org.jetbrains.annotations.NotNull;
+
+/** Project utility for various {@link Project} related utilities. */
+public final class ProjectUtil {
+
+  private static final ProjectUtil INSTANCE = new ProjectUtil();
+
+  private ProjectUtil() {}
+
+  /** Return instance of this utility. */
+  public static ProjectUtil getInstance() {
+    return INSTANCE;
+  }
+
+  /** Subscribes all open projects to the given topic with the supplied handler. */
+  public <L> void subscribeAll(@NotNull Topic<L> topic, @NotNull L handler) {
+    Stream.of(ProjectManager.getInstance().getOpenProjects())
```

Comment:
nit: Consider using `Arrays.stream()` instead of `Stream.of()` here when working with streaming array sources. One main reason is:

```
List<String> stringList = new ArrayList<String>();

// This works:
Stream.of(stringList);

// This fails to compile:
Arrays.stream(stringList);
```

While this may seem insignificant, it prevents accidental mistakes where calling a method that you *think* returns an array that actually returns a list would silently succeed, returning you a `Stream<List<String>>` instead of a `Stream<String>`.

Another reason has to do with primitive types, where `Stream.of(new int[1])` returns you a `Stream<int[]>`, but `Arrays.stream(new int[1])` returns you an `IntStream`.